strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
```

Comment:
> .onlyOne(tpr -> tpr.getClass().instanceOf MyNewTestParser.class)

This would be rather odd from an API perspective. For such cases wouldn't you instantiate `MyNewTestParser` directly instead?